omod/src/main/java/org/openmrs/web/attribute/handler/RegexValidatedTextDatatypeHandler.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.web.attribute.handler;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.customdatatype.CustomDatatype;
+import org.openmrs.customdatatype.InvalidCustomValueException;
+import org.openmrs.customdatatype.datatype.RegexValidatedTextDatatype;
+import org.springframework.stereotype.Component;
+
+/**
+ * A handler for the {@link RegexValidatedTextDatatype}.
+ */
+@Component
+public class RegexValidatedTextDatatypeHandler implements FieldGenDatatypeHandler<RegexValidatedTextDatatype, java.lang.String> {
+	
+	
+	private static final Log log = LogFactory.getLog(RegexValidatedTextDatatypeHandler.class);
+	
+	/**
+	 * @see org.openmrs.customdatatype.CustomDatatypeHandler#setHandlerConfiguration(String)
+	 */
+	@Override
+	public void setHandlerConfiguration(String arg0) {
+		// not used
+	}
+	
+	/**
+	 * @see org.openmrs.web.attribute.handler.FieldGenDatatypeHandler#getWidgetName()
+	 */
+	@Override
+	public String getWidgetName() {
+		return "java.lang.String";
```

Comment:
I don't remember how this is supposed to work, but there must be an example
of it somewhere...

-Darius (by phone)
On Jul 6, 2016 7:23 PM, "Wyclif Luyima" <notifications@github.com> wrote:

> In
> omod/src/main/java/org/openmrs/web/attribute/handler/RegexValidatedTextDatatypeHandler.java
> <https://github.com/openmrs/openmrs-module-legacyui/pull/49#discussion_r69733767>
> :
>
> > +	private static final Log log = LogFactory.getLog(RegexValidatedTextDatatypeHandler.class);
> > +	
> > +	/**
> > +	 * @see org.openmrs.customdatatype.CustomDatatypeHandler#setHandlerConfiguration(String)
> > +	 */
> > +	@Override
> > +	public void setHandlerConfiguration(String arg0) {
> > +		// not used
> > +	}
> > +	
> > +	/**
> > +	 * @see org.openmrs.web.attribute.handler.FieldGenDatatypeHandler#getWidgetName()
> > +	 */
> > +	@Override
> > +	public String getWidgetName() {
> > +		return "java.lang.String";
>
> I thought the widget name is an arbitrary value, can you confirm
> @djazayeri <https://github.com/djazayeri> ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openmrs/openmrs-module-legacyui/pull/49/files/03e0c53d56b24c43e8304e56e53469a07da2946a#r69733767>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABHHtP4cTvJ0Qj9GXev6VWl5hsS0uRKfks5qS7NQgaJpZM4JCErI>
> .
>